pkg/codegen/java/gen.go

```diff
@@ -1515,50 +1534,101 @@ func (mod *modContext) genFunctions(ctx *classFileContext, addClass addClassMeth
 			returnType = ctx.imports.Ref(names.Void)
 		}
 
+		// default method name returns Output<ReturnType>
 		methodName := tokenToFunctionName(fun.Token)
 
+		// another "plain" overload will return CompletableFuture<ReturnType>
+		plainMethodName := methodName + "Plain"
```

Comment:
Could we use the same method name (no suffix)? In theory the overload resolution would let us do that right? Wondering if that's a good idea but want to at least consider explicitly.

Reply:
In theory you could but it will not _always_ work because if you have a function without arguments, then overloads won't compile since only the return type is different:
```java
// this won't work
Output<Result> funcName();
CompletableFuture<Result> funcName();
```

Reply:
> Right. In all the other SDKs we don't generate the Output-versioned invoke if there are no arguments

> I recall what made me not worry about this case too much is that there's barely any examples of such invokes across our provider base, though they are theoretically possible ofc.

Actually I thought having both versions is much more uniform:
 - It is clear which functions return what family of types
 - It makes it easier for program-gen to assume that there is always an output-returning function invoke regardless of whether they have arguments or not, so that programgen doesn't have to wrap those with `Output.of`